### DISCUSSION TITLE
Can we run the same asset check for multiple assets?

### DISCUSSION THREAD

**Discussion by marcilj** [dagster-io/dagster#18254] [Q&A] [answered]
Based on the doc [here](https://docs.dagster.io/concepts/assets/asset-checks#asset-checks-)

It seems like asset checks need to define the asset it want to be applied to. But that asset field is not an array, so we can't put many assets for the same assets checks.

Is it possible to configure an asset checks that would do something like checks for nulls, and apply it to many assets at the same time?

**Comment by johannkm** [comment, marked as answer]
Hi @marcilj, to clarify:

Do you want to write a generic null check that gets parameterized with a set of assets, and runs independently for each asset? If so you can use [check factories](https://docs.dagster.io/concepts/assets/asset-checks#asset-check-factories).

Or would this be a check function that runs once, and returns results for many assets? That's not possible currently- the tracking issue is https://github.com/dagster-io/dagster/issues/18068
